src/rgw/rgw_http_errors.h

```diff
@@ -10,65 +10,66 @@ struct rgw_http_errors {
+    { ERR_TOO_MANY_BUCKETS, 400, "TooManyBuckets", "You have attempted to create more buckets than allowed." },
+    { ERR_MALFORMED_XML, 400, "MalformedXML", "This happens when the user sends malformed xml (xml that doesn't conform to the published xsd) for the configuration. The error message is, \"The XML you provided was not well-formed or did not validate against our published schema.\"" },
+    { ERR_AMZ_CONTENT_SHA256_MISMATCH, 400, "XAmzContentSHA256Mismatch", "Amazon contents sha256 mismatch"},
+    { ERR_LENGTH_REQUIRED, 411, "MissingContentLength", "You must provide the Content-Length HTTP header." },
+    { EACCES, 403, "AccessDenied", "Access Denied" },
+    { EPERM, 403, "AccessDenied", "Access Denied" },
+    { ERR_SIGNATURE_NO_MATCH, 403, "SignatureDoesNotMatch", "The request signature we calculated does not match the signature you provided. Check your AWS secret access key and signing method. For more information, see REST Authentication and SOAP Authentication for details." },
+    { ERR_INVALID_ACCESS_KEY, 403, "InvalidAccessKeyId", "The AWS access key Id you provided does not exist in our records." },
+    { ERR_USER_SUSPENDED, 403, "UserSuspended", "User Suspended" },
+    { ERR_REQUEST_TIME_SKEWED, 403, "RequestTimeTooSkewed", "The difference between the request time and the server's time is too large." },
+    { ERR_QUOTA_EXCEEDED, 403, "QuotaExceeded", "Quota Exceeded" }, 
+    { ENOENT, 404, "NoSuchKey", "The resource you requested does not exist" },
+    { ERR_NO_SUCH_BUCKET, 404, "NoSuchBucket", "The specified bucket does not exist." },
+    { ERR_NO_SUCH_WEBSITE_CONFIGURATION, 404, "NoSuchWebsiteConfiguration", "No such website configuration" },
+    { ERR_NO_SUCH_UPLOAD, 404, "NoSuchUpload", "The specified multipart upload does not exist. The upload ID might be invalid, or the multipart upload might have been aborted or completed." },
+    { ERR_NOT_FOUND, 404, "Not Found", "Not Found"},
+    { ERR_NO_SUCH_LC, 404, "NoSuchLifecycleConfiguration", "The lifecycle configuration does not exist." },
+    { ERR_METHOD_NOT_ALLOWED, 405, "MethodNotAllowed", "The specified method is not allowed against this resource." },
+    { ETIMEDOUT, 408, "RequestTimeout", "Your socket connection to the server was not read from or written to within the timeout period." },
+    { EEXIST, 409, "BucketAlreadyExists", "The requested bucket name is not available. The bucket namespace is shared by all users of the system. Please select a different name and try again." },
+    { ERR_USER_EXIST, 409, "UserAlreadyExists", "The requested user name is not available. Please select a different name and try again." }, // doesn't exist in s3 error codes
+    { ERR_EMAIL_EXIST, 409, "EmailExists", "The request email is not available. Please select a different email and try again." },
+    { ERR_KEY_EXIST, 409, "KeyExists", "The request key is not available. Please select a different key and try again." },
```

Comment:
None of `UserAlreadyExists`, `EmailExists`, or `KeyExists` are listed in Amazon's documentation as error response codes (https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList), so generic S3 clients are not likely to understand any of them. Should this be documented as a comment for all three of these lines rather than just the first one?